mempool/mempool.go

```diff
@@ -478,6 +478,11 @@ func (cache *txCache) Push(tx types.Tx) bool {
 	cache.mtx.Lock()
 	defer cache.mtx.Unlock()
 
+	// if cache size is 0, do nothing
+	if cache.size == 0 {
```

Comment:
Wouldn't it be more correct to create an interface and `nopCache{}`, which does nothing. Then in the constructor we can have

```go
if config.CacheSize > 0 {
  txCache = newTxCache()
} else {
  txCache = nopCache{}
}
```

Reply:
yes, thank you. want to do it and just push here ?

Reply:
push into release branch... umm, ok

Reply:
done